### PR TITLE
revert change to quad selection (resolves #7025)

### LINKED ISSUE #7025: Discussion on quad point selection
As mentioned by Pipou on discord, before #6844 it was easy to select the same corners on multiple quads. You just had to select multiple quads using shift + drag and then every corner would be selected on all quads. I agree that this seems to be a very useful feature which got removed. There are a few ways I could think of to solve this:

1. Add a shortcut to use the old selection style. So when some corners are selected on one quad you could press alt + click to select the quads on which these corners also should be selected.
2. Add an option (under the settings tab) to select which style of selection is preferred.
3. Revert the changes of #6844. Since points are now easier to select the change might not be that useful after all.

Right now I'm leaning towards option 3 because I'm not sure I like the way selection in tile layers and quad layers differs now. That is selection automatically copies on tile layers but ctrl+c/ctrl+v is required on quad layers. However the new selection style seems a bit more intuitive and more flexible.

### PR DESCRIPTION
**Pull request by Marmare314**
Some refactorings are kept and rotation being bound to `R` is also not reverted.

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by heinrich5991** [comment]
```
src/game/editor/popups.cpp:1255:6: error: 'auto pQuad' can be declared as 'auto *pQuad' [readability-qualified-auto,-warnings-as-errors]
        for(auto pQuad : vpQuads)
            ^~~~~
            auto *
```

File: src/game/editor/editor.cpp [hunk 1]
```diff
@@ -2531,31 +2445,16 @@ void CEditor::DoMapEditor(CUIRect View)
...
-
 							for(size_t i = 0; i < pQuadLayer->m_vQuads.size(); i++)
 							{
-								CQuad *pQuad = &pQuadLayer->m_vQuads[i];
-
-								for(size_t v = 0; v < 5; v++)
-								{
-									float px = fx2f(pQuad->m_aPoints[v].x);
-									float py = fx2f(pQuad->m_aPoints[v].y);
+								const CQuad &pQuad = pQuadLayer->m_vQuads[i];
```
Comment (Robyt3) [inline src/game/editor/editor.cpp:2452]: `pQuad` -> `Quad`

**Comment by Marmare314** [comment]
Thanks, is there any way to run the clang-tidy check locally too? I currently only check `fix-style.py` before pushing.

**Comment by Robyt3** [comment]
> Thanks, is there any way to run the clang-tidy check locally too?

If you use Visual Studio then there's probably integration for that and it's probably a lot easier to use than the other method.

If you use MSYS2 then install the relevant clang packages (I think `mingw-w64-clang-x86_64-clang`, `mingw-w64-clang-x86_64-clang-analyzer`, `mingw-w64-clang-x86_64-clang-tools-extra` should be enough) and run basically the commands that the workflow uses:

https://github.com/ddnet/ddnet/blob/a07fd521f75e0ef638eb1522b9a391d91870462b/.github/workflows/clang-tidy.yml#L29-L32

**Comment by Marmare314** [comment]
Would be cool if there was a script for that too. And one which executes all CI checks (except the ones only checking if the source compiles with specific compilers).

**Pull request merged**